src/main/kotlin/org/opensearch/commons/alerting/model/ChainedFindings.kt

```diff
@@ -0,0 +1,76 @@
+package org.opensearch.commons.alerting.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.notifications.model.BaseModel
+import org.opensearch.commons.utils.validateId
+import java.io.IOException
+
+/**
+ * Context passed in delegate monitor to filter data queried by a monitor based on the findings of the given monitor id.
+ */
+data class ChainedFindings(
```

Comment:
Make this `ChainedInput` and implement the [Input class](https://github.com/opensearch-project/common-utils/blob/main/src/main/kotlin/org/opensearch/commons/alerting/model/Input.kt)